### PR TITLE
Support for SMB2

### PR DESCRIPTION
**Pull request by ibauersachs**
Replace jcifs with smbj to support SMB2.

While this code works for us, the PR isn't really ready to merge. I'm opening this so that someone can take it from here.

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -29,19 +29,20 @@
...
-        <version>1.532.1</version>
+        <version>2.9</version>
     </parent>
 
     <groupId>org.jenkins-ci.plugins</groupId>
     <artifactId>publish-over-cifs</artifactId>
     <packaging>hpi</packaging>
     <name>Publish Over CIFS</name>
-    <version>0.4-SNAPSHOT</version>
+    <version>0.5-xovis3-SNAPSHOT</version>
```
Comment (slide) [inline pom.xml]: This needs to be reverted.

**Comment by slide** [comment]
A major concern here is backward compatibility, have you tested (or written tests which would be even better) for upgrading from a previous installation to this version?

**Comment by ibauersachs** [comment]
No, I haven't written tests to upgrade from an older plugin version, but since I didn't touch anything there I wouldn't expect it to break. Regarding backwards compatibility I'd be more concerned about people who actually want SMBv1 in legacy infrastructures. It might be better to create a new plugin from this PR (i.e. publish-over-smb-plugin).

Regarding the pom-version string, sure, as I said, I wouldn't recommend this for primetime. But if it's the only thing preventing you from merging it, I'll remove it.

**Comment by slide** [comment]
For backward compatibility, I was referring to the settings for the plugin, not smbv1. I'll pull this on my machine and do some upgrade testing. Thanks for the work!

**Comment by mklemarczyk** [comment]
Hi @ibauersachs can you share your build of this plugin? I can not use older versions of SMB dude to Multiple security vulnerabilities.

**Comment by jameswragg** [comment]
Any news on this? After some internal updates to our server Publish Over CIFS has died (test connection fails). I'm assuming it's lack of SMB2 support, though not certain.

**Comment by jameswragg** [comment]
Aah commented too quickly - I see commits around SMB2 support (yay!). So the question is, when will that be released? Thanks.

**Comment by slide** [comment]
I am working on incorporating jcifs-ng which is a lot less changes to the overall plugin. Hopefully should be released soon.

**Pull request closed without merging**